### ISSUE TITLE
Change Request: Extract reference identifier normalization into a shared utility

### ISSUE THREAD

**Issue by KumJungMin** [eslint/markdown#731]
### Environment

ESLint version: v10.7.0
@eslint/markdown version: 8.0.3
Node version: v22.18.0
npm version: v10.9.3
Operating System: darwin 24.6.0

### What problem do you want to solve?

The following four rules currently use `normalizeIdentifier(value).toLowerCase()` in a total of seven places to compare reference identifiers:
- no-duplicate-definitions
- no-unused-definitions
- no-empty-definitions
- no-reference-like-urls

<img width="515" height="480" alt="Image" src="https://github.com/user-attachments/assets/8920b5e9-4393-430a-a29c-d0989a720999" />

This expression converts identifiers in allowlists or link destinations into a form that can be compared with the AST’s identifier. 
These rules share the same comparison behavior, but each call site handles normalization and lowercase conversion separately.

The current behavior is correct. 
I’d like to make the purpose of this shared transformation explicit and make it easier to keep the rules consistent 
when updating the related code.

### What do you think is the correct solution?

I’d suggest adding a `normalizeReferenceIdentifier()` function to `src/util.js` and replacing the repeated expression with calls to that function.

```js
export function normalizeReferenceIdentifier(value) {
    return normalizeIdentifier(value).toLowerCase();
}
```
This would preserve the existing operations and their order, with no changes to rule options, diagnostic messages, or lint results. 
The existing tests for the affected rules can help verify that case handling, whitespace normalization, 
and Unicode identifier handling remain unchanged.

### Participation

- [x] I am willing to submit a pull request for this change.

### AI acknowledgment

- [ ] I did not use AI to generate this issue report.
- [x] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

> Disclosure: I'm a participant of [open source contribution program OSSCA](https://github.com/eslint-ossca)

**Comment by lumirlumir** [comment]
> Disclosure: I'm a participant of [open source contribution program OSSCA](https://github.com/eslint-ossca): confirmed.

Thanks for the issue. I think extracting it into a shared utility makes sense. I also think using the original name, `normalizeIdentifier`, rather than `normalizeReferenceIdentifier`, would be more appropriate, since the mdast specification uses `identifier` as the property name for this use case.

If you’re concerned about a name collision in the implementation, we could use `import { normalizeIdentifier as micromarkUtilNormalizeIdentifier } from 'micromark-util-normalize-identifier'` or something similar to avoid the collision.
